src/Microsoft.Health.Dicom.Functions/Indexing/ReindexDurableFunction.Activity.cs

```diff
@@ -128,5 +128,23 @@ public async Task ReindexInstancesAsync([ActivityTrigger] ReindexInstanceInput i
 
             await Task.WhenAll(tasks);
         }
+
+        /// <summary>
+        /// Fetch schema version.
+        /// </summary>
+        /// <param name="context">The durable activity context.</param>
+        /// <param name="log">The log</param>
+        /// <returns>The task.</returns>
+        [FunctionName(nameof(FetchSchemaVersionAsync))]
+        public async Task FetchSchemaVersionAsync([ActivityTrigger] IDurableActivityContext context, ILogger log)
+        {
+            // TODO: performance improvement, don't need to call service for every call.
+            EnsureArg.IsNotNull(context, nameof(context));
+            EnsureArg.IsNotNull(log, nameof(log));
+            log.LogInformation("Fetching schema version");
+            int version = await _schemaManagerDataStore.GetCurrentSchemaVersionAsync(default);
+            _schemaInformation.Current = version;
```

Comment:
We most likely need to register our use of this schema version in the database also. What happens if this is running and the schema version is upgraded under the hood? Ideally we would inform the database of our use of the schema along with the min/max that it supports.

Reply:
Yes, that's something Will and I discuss about. 
Reindex job could take very long time, what if database schema is upgrade in the between? 
In Dicom Web Server, we always have latest binary, while in Azure Function, we could have old version. 
How should we handle with that is still an open question. 
So I'm thinking maybe we should start from simplest and slowly improve it.
Btw, Will has another PR to simplify the way to get version https://github.com/microsoft/dicom-server/pull/843